pkg/cri/cri.go

```diff
@@ -0,0 +1,66 @@
+package cri
+
+import (
+	"context"
+	"time"
+
+	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
+	runtimeapi "k8s.io/cri-api/pkg/apis/runtime/v1"
+	k8sutil "k8s.io/kubernetes/pkg/kubelet/util"
+)
+
+const (
+	maxMsgSize   = 1024 * 1024 * 16
+	socketPrefix = "unix://"
+)
+
+// WaitForCRIService blocks in a retry loop until the CRI service
+// is functional at the provided socket address. It will return only on success,
+// or when the context is cancelled.
+func WaitForCRIService(ctx context.Context, address string, service string) error {
```

Comment:
Additional nit: to avoid stuttering, this could probably just be called WaitForService now that it's in a `cri` package.
```suggestion
func WaitForService(ctx context.Context, address string, service string) error {
```